### DISCUSSION TITLE
Vim: Command name for search backwards / reverse search / `?`

### DISCUSSION THREAD

**Discussion by thomasheartman** [zed-industries/zed#22164] [Feature Requests]
Hey!

What's the comand called that's mapped to `?` in Vim's normal mode (it searches backwards)? I can't seem to find it.

I'm coming from a Emacs/Evil background and trying to get Zed up and running. Through a long time using a mix of Emacs, Vim, and custom key keybindings, I have come to rely on "ctrl-r" as a reverse search (that is: Vim's `?` in normal mode). So I wanted to set it up so that it'd work ilke that in Zed, but I can't seem to find the name of the command that does it.

For `/`, it's `vim::Search` (to which I've also added the binding `ctrl-s`), but I can't seem to find out what the binding is called for `?`.

Semi-related, is there an "explain what this key binding does" in Zed? Again, coming from Emacs, I'm used to being able to ask Emacs what a keybinding does (by "ctrl-h" followed by the keybinding in question). It's super convenient and would have made finding this out a piece of cake. Does Zed have something like that?

**Comment by 0x2CA** [comment]
```json
      "/": "vim::Search",
      "g /": "pane::DeploySearch",
      "?": ["vim::Search", { "backwards": true }],
```

![image](https://github.com/user-attachments/assets/9f076ead-cbac-40ec-a1aa-ad1a2d509daf)

**Reply by thomasheartman** [reply]
Yeah, thanks! I found it myself in the end, but this is the right one 😄 Wasn't obvious to me how I was supposed to find it, so it was a stroke of luck.

Is there no way to "search action by key binding" in Zed?

**Reply by 0x2CA** [reply]
> Yeah, thanks! I found it myself in the end, but this is the right one 😄 Wasn't obvious to me how I was supposed to find it, so it was a stroke of luck.
> 
> Is there no way to "search action by key binding" in Zed?

There's no such feature right now, but it's certainly something the zed team is planning for

**Reply by 0x2CA** [reply]
However, there is a view that checks the keyboard context

![image](https://github.com/user-attachments/assets/d624e554-e534-4cbf-bb29-811cdd3382c7)

**Reply by thomasheartman** [reply]
Yes; thanks! Just found that this morning (I'm not great at checking my github notifs before running off into the docs 😅). Yeah, that one works perfectly well for what I'm doing.

And for posterity: the command to invoke it is `debug::OpenKeyContextView`, which I've bound to `ctrl-h` to aid my muscle memory.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
